drivers/net/nsos_sockets.c

```diff
@@ -0,0 +1,947 @@
+/**
+ * Copyright (c) 2023-2024 Marcin Niestroj
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *
+ * Zephyr (top) side of NSOS (Native Simulator Offloaded Sockets).
+ */
+
+#define _POSIX_C_SOURCE 200809
```

Comment:
Better undef it before, just to avoid a possible redefinition warning if somebody happens to define it globally.
(undefining is safe even if it was never set)

Reply:
Added `#undef _POSIX_C_SOURCE` and changed value to `200809L`.